### PR TITLE
Week 4 Features

### PR DESCRIPTION
**Pull request by TBarrett553**
-A chart has been added
-Rearranged some styling components
-delete comments feature has been refined

Todo
-sentiment analysis (will probably be developed in week 5)

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -29,12 +30,19 @@
...
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        //Check if the quantity is null. This usually happens when the user leave the quantity input blank
+
+        if (request.getParameter("quantity") == null) {
+            Integer quantity = 5;
```
Comment (freeboson) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: This has no effect. If quantity is null, it creates a temporary variable and then deletes it.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -29,12 +30,19 @@
...
+        }
+
         Integer quantity = Integer.parseInt(request.getParameter("quantity"));
```
Comment (freeboson) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: this should be e.g. 

```
// (define kDefaultQuantity = 5 somewhere above)
Integer quantity = request.getParameter("quantity") != null ? Integer.parseInt(...) : kDefaultQuantity;
```

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -67,3 +70,25 @@ function deleteComments() {
     comments.innerHTML = "";
     console.log("messages gone")
 }
+
+/** Creates a chart and adds it to the page. */
+function drawChart() {
+  const data = new google.visualization.DataTable();
+  data.addColumn('string', 'Result');
+  data.addColumn('number', 'Percent');
+        data.addRows([
```
Comment (flying-bus) [inline portfolio/src/main/webapp/script.js:79]: indentation is weird here.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (TBarrett553) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: Good to know, I'll just remove that conditional.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -19,5 +19,7 @@
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     throw new UnsupportedOperationException("TODO: Implement this method.");
+
```
[the remark this answers is not in the document]
Comment (TBarrett553) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:22]: Yes, the actual content will be in the next branch. Thank you for the guidance on this!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (freeboson) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: What does `Integer.parseInt(null)` do? set it to zero? Does that work for you?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -37,10 +37,10 @@
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-        // If the quantity is null it will be covered. Null values will happen when the user leave the quantity input blank. 
-
-        Integer quantity = Integer.parseInt(request.getParameter("quantity"));
-
+        // If the quantity is null it will be covered by the default qaunity. Null values will happen when the user leave the quantity input blank. 
+        int defaultQuantity = 5;
```
Comment (freeboson) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:41]: nit: make this a constant, and follow style guide `DEFAULT_QUANTITY` (https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names)

**Pull request merged**